core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/DataWritingCommandExecParser.scala

```diff
@@ -28,16 +28,111 @@ case class DataWritingCommandExecParser(
+      finalSpeedup, duration, node.id, writeSupported, None)
   }
 }
+
+// A case class used to hold information of the parsed Sql-node.
+// The purpose is to be able to reuse the code between different components.
+case class DataWritingCmdWrapper(
+    execName: String,
+    mappedExec: String,
+    dataFormat: String)
+
+object DataWritingCommandExecParser {
+  private val dataWriteCMD = "DataWritingCommandExec"
+  private val defaultPhysicalCMD = dataWriteCMD
+  // made public for testing
+  val saveIntoDataSrcCMD = "SaveIntoDataSourceCommand"
+  val insertIntoHadoopCMD = "InsertIntoHadoopFsRelationCommand"
+
+  // List of writeExecs that represent a physical command.
+  // hardcode because InsertIntoHadoopFsRelationCommand uses this same exec
+  // and InsertIntoHadoopFsRelationCommand doesn't have an entry in the
+  // supported execs file
+  private val physicalWriteCommands = Set(
+    defaultPhysicalCMD
+  )
+
+  // A set of the logical commands that will be mapped to the physical write command
+  // which has an entry in the speedupSheet
+  // - InsertIntoHadoopFsRelationCommand is a logical command that does not have an entry in the
+  //   supported execs file
+  // - SaveIntoDataSourceCommand is generated by DeltaLake Table writes
+  private val logicalWriteCommands = Set(
+    dataWriteCMD,
+    insertIntoHadoopCMD,
+    saveIntoDataSrcCMD
+  )
+
+  // Defines a list of the execs that include formatted data.
+  // This will be used to extract the format and then check whether the
+  // format is supported or not.
+  private val formattedWriteCommands = Set(
+    dataWriteCMD,
+    insertIntoHadoopCMD
+  )
+
+  // For now, we map the SaveIntoDataSourceCommand to defaultPhysicalCMD because we do not
+  // have speedup entry for the deltaLake write operation
+  private val logicalToPhysicalCmdMap = Map(
+    insertIntoHadoopCMD -> defaultPhysicalCMD,
+    saveIntoDataSrcCMD -> defaultPhysicalCMD
+  )
+
+  // Map to hold the relation between writeExecCmd and the format.
+  // This used for expressions that do not show the format as part of the description
+  private val specialWriteFormatMap = Map(
+    saveIntoDataSrcCMD -> "Delta"
+  )
+
+  // Checks whether a node is a write CMD Exec
+  def isWritingCmdExec(nodeName: String): Boolean = {
+    logicalWriteCommands.exists(nodeName.contains(_))
+  }
+
+  def getWriteCMDWrapper(node: SparkPlanGraphNode): Option[DataWritingCmdWrapper] = {
+    val processedNodeName = node.name.trim
+    logicalWriteCommands.find(processedNodeName.contains(_)) match {
+      case None => None
+      case Some(wCmd) =>
+        // get the dataformat from the map if it exists.
+        // Otherwise, fallback to the string parser
+        val dataFormat = specialWriteFormatMap.get(wCmd) match {
+          case Some(f) => f
+          case None => getWriteFormatString(node.desc)
+        }
+        val physicalCmd = logicalToPhysicalCmdMap.getOrElse(wCmd, defaultPhysicalCMD)
```

Comment:
Nit: I don't think default `defaultPhysicalCMD` is required as we are already checking if the node is a write cmd in `isWritingCmdExec`. So wCmd should be one of the parameters in `logicalWriteCommands`.